app/services/RestaurantService.java

```diff
@@ -254,4 +279,20 @@ public String updatePicture(final ImageUploadForm imageUploadForm) throws Except
 		getSession().update(restaurant);
 		return "{ \"imageFor\": \"" + imageUploadForm.getImageType() + "\", \"url\": \"" + newImagePath + "\"}";
 	}
+
+	private Boolean getRatingClass(int filter, double averageRating) {
+		if(averageRating >= 0.25 && filter <= 1) {
+			return true;
+		} else if(averageRating >= 2 && filter <= 2) {
+			return true;
+		} else if(averageRating >= 3 && filter <= 3) {
+			return true;
+		} else if(averageRating >= 4 && filter <= 4) {
+			return true;
+		} else if(averageRating >= 4.75 && filter <= 5) {
+			return true;
+		} else {
+			return false;
+		}
```

Comment:
Incredibly difficult to read/understand. Try finding a more descriptive name for the method and consider changing the logic of the method.

Try something like this:
```
if (averageRating >= 0.25) {
  if (filter == 5) {
    return averageRating >= 4.75;
  }

  return averageFilter >= filter;
}

return false;
```
This might also work for you, but it changes the result.
```
  return averageRating >= 0.25 && averageFilter >= filter - 0.25;
```